### ISSUE TITLE
build failed with missing header

### ISSUE THREAD

**Issue by tkclimb** [tensorflow/runtime#5]
Hi, 
I'm recently tried to build TFRT by following the [installation guide](https://github.com/tensorflow/runtime#prerequisites).
However, I found failure with errors after starting `bazel build -c opt //tools:bef_executor`, so I put the message I got with `--verbose_failures` flag.
It seems clang couldn't find some headers such as `stdbuf.h`. I think they are normally installed in `/usr/include/linux/`, `/usr/lib/gcc/x86_64-linux-gnu/<your version>/include/`. ofcourse you can manually install with `libc++` or `libclang-common-dev` (but they are not one with libstdc++).
I'm not very sure using them is proper or not, is there anyone who knows about it??

I just in case pasted Dockerfile I used to build the env as well for some help.

```shell
root@22232bd2d0ae:/tfrt# bazel build -c opt //tools:bef_executor
WARNING: /root/.cache/bazel/_bazel_root/738a6075dc23ce42c0be9b4ee7a7a5da/external/zlib/BUILD.bazel:5:1: in includes attribute of cc_library rule @zlib//:zlib: ignoring invalid absolute path '/usr/lib/gcc/x86_64-linux-gnu/9/include'
INFO: Analyzed target //tools:bef_executor (29 packages loaded, 4054 targets configured).
INFO: Found 1 target...
ERROR: /root/.cache/bazel/_bazel_root/738a6075dc23ce42c0be9b4ee7a7a5da/external/zlib/BUILD.bazel:5:1: C++ compilation of rule '@zlib//:zlib' failed (Exit 1) clang failed: error executing command /usr/bin/clang -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 25 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from external/zlib/deflate.c:52:
In file included from external/zlib/deflate.h:16:
In file included from external/zlib/zutil.h:22:
In file included from external/zlib/zlib.h:34:
external/zlib/zconf.h:247:14: fatal error: 'stddef.h' file not found
#    include <stddef.h>
             ^~~~~~~~~~
1 error generated.
Target //tools:bef_executor failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 16.017s, Critical Path: 0.09s
INFO: 4 processes: 4 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

The Dockerfile used to build the env.
```
FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu16.04

ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
ENV DEBIAN_FRONTEND noninteractive
ENV TZ=Asia/Tokyo

ARG workspace
ENV workspace /tfrt
WORKDIR $workspace

RUN echo $TZ > /etc/timezone && \
  apt-get update && apt-get install -y tzdata && \
  rm /etc/localtime && \
  ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && \
  dpkg-reconfigure -f noninteractive tzdata && \
  apt-get clean

RUN apt-get update --fix-missing && apt-get install -y \
  software-properties-common wget bzip2 git

RUN add-apt-repository -y ppa:ubuntu-toolchain-r/test && \
  apt-get update && \
  apt-get install -y gcc-8 g++-8

RUN wget -q https://github.com/bazelbuild/bazelisk/releases/download/v0.0.8/bazelisk-linux-amd64 && \
  mv bazelisk-linux-amd64 /usr/local/bin/bazel && \
  chmod +x /usr/local/bin/bazel

RUN wget -q https://apt.llvm.org/llvm.sh -O /tmp/llvm.sh && \
  chmod +x /tmp/llvm.sh && \
  /tmp/llvm.sh 9

RUN update-alternatives --install /usr/bin/clang clang /usr/bin/clang-9 9 && \
  update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-9 9
```

**Comment by byronyi** [comment]
You need clang-9.0.

**Comment by bhack** [comment]
```
clang++ -v |& grep "Selected GCC"
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/8
```
https://github.com/tensorflow/runtime#install-libstdc

**Comment by bhack** [comment]
@tkclimb With `FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04` seems it is compiling fine.

**Comment by tkclimb** [comment]
@byronyi 
I think from `clang-8`,  we don't need the miner version like it was like `clang-6.0`.
also, the clang compiler seems properly available from the shell in the docker environment.

@bhack 
I already tried the command and it's correct.
```shell
root@22232bd2d0ae:/tfrt# clang++ -v |& grep "Selected GCC"
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/8
```

I saw the document says TFRT currently officially supports only ubuntu16 in [here](https://github.com/tensorflow/runtime#getting-started) is it ok for that??
(it's just a question, I'll try ubuntu18 anyway)
